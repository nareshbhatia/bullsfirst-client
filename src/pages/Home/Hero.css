.hero_container {
  position: relative;
}

.hero__slogan.title {
  line-height: 32px;
  max-width: 392px;
  margin-top: var(--spacing-5);
}

.hero__popsicle {
  display: none;
  position: absolute;
  right: 200px;
  bottom: -150px;
  transform: rotate(25deg);
  filter: drop-shadow(2px 8px 2px rgba(0, 0, 0, 0.06))
    drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.35));
}
@media (min-width: 640px) {
  .hero__popsicle {
    display: block;
  }
}
