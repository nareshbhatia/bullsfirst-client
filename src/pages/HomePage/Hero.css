.hero_container {
  position: relative;
}

.hero__slogan {
  margin: var(--spacing-5) 0 0 0;
  font-size: 30px;
  line-height: 32px;
  max-width: 392px;
}

.hero__popsicle {
  display: none;
  position: absolute;
  right: 200px;
  bottom: -150px;
  transform: rotate(25deg);
  filter: drop-shadow(2px 8px 2px rgba(0, 0, 0, 0.06))
    drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.35));
}
@media (min-width: 640px) {
  .hero__popsicle {
    display: block;
  }
}
