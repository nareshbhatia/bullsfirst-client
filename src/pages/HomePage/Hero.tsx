import React from 'react';
import { useNavigate } from 'react-router-dom';
import logo from '../../assets/bullsfirst-logo.svg';
import popsicle from '../../assets/popsicle.png';
import { HorizontalContainer } from '../../components';
import './Hero.css';

export const Hero = () => {
  const navigate = useNavigate();

  /* istanbul ignore next */
  const handleSignIn = async () => {
    navigate('/signin');
  };

  return (
    <div data-testid="home-hero" className="bg-neutral-light py-4">
      <div className="hero_container container">
        <HorizontalContainer className="justify-between items-center">
          <img src={logo} alt="logo" width="190" />
          <img
            src={popsicle}
            alt="popsicle"
            height="200px"
            className="hero__popsicle"
          />
          <button
            className="btn-sm mt-2"
            aria-label="Sign in"
            type="submit"
            onClick={handleSignIn}
          >
            Sign in
          </button>
        </HorizontalContainer>
        <p className="hero__slogan">
          Get better results with Bullsfirst at the helm of your portfolio.
        </p>
      </div>
    </div>
  );
};
