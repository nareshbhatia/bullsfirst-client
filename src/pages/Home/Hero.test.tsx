import React from 'react';
import { render } from '../../test/test-utils';
import { Hero } from './Hero';

describe('<Hero />', () => {
  test('renders correctly', () => {
    const { asFragment } = render(<Hero />);
    expect(asFragment()).toMatchSnapshot();
  });
});
