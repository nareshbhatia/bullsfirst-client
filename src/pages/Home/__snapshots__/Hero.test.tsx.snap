// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="bg-neutral-light py-4"
    data-testid="home-hero"
  >
    <div
      class="hero_container container"
    >
      <div
        class="flex-1 flex flex-row justify-between items-center"
        data-testid="horizontal-container"
      >
        <img
          alt="logo"
          src="bullsfirst-logo.svg"
          width="190"
        />
        <img
          alt="popsicle"
          class="hero__popsicle"
          height="200px"
          src="popsicle.png"
        />
        <button
          aria-label="Sign in"
          class="btn-sm mt-2"
          type="submit"
        >
          Sign in
        </button>
      </div>
      <p
        class="title hero__slogan"
      >
        Get better results with Bullsfirst at the helm of your portfolio.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
